cmd/ebrelayer/txs/parser.go

```diff
@@ -0,0 +1,57 @@
+package txs
+
+// --------------------------------------------------------
+//      Parser
+//
+//      Parses structs containing event information into
+//      unsigned transactions for validators to sign, then
+//      relays the data packets as transactions on the
+//      Cosmos Bridge.
+// --------------------------------------------------------
+
+import (
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/peggy/cmd/ebrelayer/events"
+	ethbridgeTypes "github.com/cosmos/peggy/x/ethbridge/types"
+)
+
+// ETH : ETH constant specifies a token type of Ethereum
+const (
+	ETH string = "eth"
+)
+
+// ParsePayload : ParsePayload parses a LockEvent struct, packaging the information
+//				  along with a validator address in an EthBridgeClaim msg
+func ParsePayload(valAddr sdk.ValAddress, event *events.LockEvent) (ethbridgeTypes.EthBridgeClaim, error) {
+
+	witnessClaim := ethbridgeTypes.EthBridgeClaim{}
+
+	// Nonce type casting (*big.Int -> int)
+	nonce := int(event.Nonce.Uint64())
```

Comment:
if it is being casted to `int` why not use `event.Nonce.Int64()`?

Reply:
Good catch. Fixed https://github.com/musnit/peggy/pull/6/commits/0b7406c917b4cdfe0d069af0f49a22157bedf154